realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1524,6 +1524,18 @@ public boolean isValid() {
         this.query.or();
         return this;
     }
+    
+    /**
+     * Logical-and two conditions
+     * Technically, does nothing. Intended purely for syntactic sugar
```

Comment:
It would rewrite this to: "Realm automatically applies logical-and between all query statements, so this is intended only as a mean to increase readability."